src/ShopBundle/objects/db/TShopCategory.class.php

```diff
@@ -486,29 +486,37 @@ protected function NumberOfArticlesInCategory($bRecursive = false)
      */
     public function AllowDisplayInShop()
     {
-        $bShowCat = false;
-
-        $oShop = TdbShop::GetInstance();
-        if (!is_null($oShop) && !$oShop->fieldShowEmptyCategories) {
-            $bShowCat = ($this->GetNumberOfArticlesInCategory(true) > 0);
-        } else {
-            $bShowCat = true;
+        if (true === $this->isEmptyAndShouldNotBeShown()) {
```

Comment:
when a method requires an "and" in its name, than I would suggest splitting it into two as it does more than one thing

Reply:
done